### ISSUE TITLE
Add a single item from another thread

### ISSUE THREAD

**Issue by neimanpinchas** [MintPlayer/MintPlayer.Dotnet.Tools#8]
I am trying to use your lib to circumvent the issue with WPF being unable to add an item from a diffrent thread.

```
// If there's only one item being added/removed, there's no need to do anything special.
            if (!IsRange(e))
            {
                base.OnCollectionChanged(e);
            }
```

as far as I understand from that if, is that adding a single item will be done on the thread calling Add, Iam I wrong?

**Comment by PieterjanDeClippel** [comment]
Nope, on the main thread that's updating the ui. Or that created the `ObservableCollection` to be more precise

**Comment by neimanpinchas** [comment]
I am sorry that I was tired and wasn't that clear, I've also struggled to understand my own message.

Who is delegating the `base.OnCollectionChanged(e);` to the responsible thread?

given this SO discussion, the event handlers are being executed on the firing thread.
https://stackoverflow.com/questions/2459634/in-net-what-thread-will-events-be-handled-in

I've hav'nt made any small case testing on it, but I've received the WPF error on Adding a single item.

Can you please explain where the correct thread will kick in?

**Comment by PieterjanDeClippel** [comment]
Example:

- An `ObservableCollection` is being constructed [here](https://github.com/PieterjanDeClippel/WpfRangeOperations/blob/master/WpfRangeOperations/MainVM.cs#L22)
- The `ObservableCollection` remembers this `SynchronizationContext`
- [Here](https://github.com/PieterjanDeClippel/WpfRangeOperations/blob/master/WpfRangeOperations/MainVM.cs#L76-L78) you see an example of items being added from another thread
- Each time an event is to be fired, it will [run](https://github.com/MintPlayer/MintPlayer.ObservableCollection/blob/master/MintPlayer.ObservableCollection/ObservableCollection.cs#L246) in this `SynchronizationContext`
